pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,312 @@
+package cgroup
+
+import (
+	"bufio"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/aquasecurity/tracee/pkg/mount"
+)
+
+//
+// Constants
+//
+
+const procCgroups = "/proc/cgroups"
+const procFilesystems = "/proc/filesystems"
+
+const CgroupV1FsType = "cgroup"
+const CgroupV2FsType = "cgroup2"
+
+const DefaultCgroupController = "cpuset"
+
+const TmpCgroup = "/tmp"
+const TmpCgroupV1Path = "tracee-cgroupv1"
+const TmpCgroupV2Path = "tracee-cgroupv2"
+
+type CgroupVersion int
+
+const (
+	CgroupV1 CgroupVersion = iota + 1
+	CgroupV2
+)
+
+//
+// Cgroups
+//
+
+// Cgroups handles cgroups initialization (v1+v2 or v2 only)
+type Cgroups struct {
+	cgroupv1 *Cgroup // used for cgroup filesystem readings
+	cgroupv2 *Cgroup // used for cgroup prog attachment (and readings, if v2 is default)
+	cgroup   *Cgroup // default cgroup (might point to one or another)
+}
+
+// NewCgroups tries to initialize both: cgroupv1 and cgroupv2, setting one as
+// the default depending on the running environment
+func NewCgroups() (*Cgroups, error) {
+	cs := &Cgroups{}
+
+	isCgroupV1, err := IsCgroupV1Default() // initialize cgroupv1
+	if err != nil {
+		return cs, err
+	}
+	if isCgroupV1 {
+		cs.cgroupv1, err = NewCgroup(CgroupV1)
+		if err != nil {
+			return cs, err // cgroupv1 is unavailable when it should be
+		}
+		cs.cgroup = cs.cgroupv1 // default is cgroupv1
+	}
+
+	isCgroupV2, err := IsCgroupV2Supported() // initialize cgroupv2
+	if err != nil {
+		return cs, err
+	}
+	if !isCgroupV2 {
+		return cs, failCgroupV2NotSupported()
+	}
+
+	cs.cgroupv2, err = NewCgroup(CgroupV2)
+	if err != nil {
+		return cs, err // cgroupv2 is unavailable (needed for cgroup progs)
+	}
+
+	if cs.cgroup == nil {
+		cs.cgroup = cs.cgroupv2 // default is cgroupv2
+	}
+
+	return cs, nil
+}
+
+// GetDefault returns the default cgroup being used by running OS
+func (cs *Cgroups) GetDefault() *Cgroup {
+	return cs.cgroup
+}
+
+func (cs *Cgroups) GetV1() *Cgroup {
+	return cs.cgroupv1
+}
+
+func (cs *Cgroups) GetV2() *Cgroup {
+	return cs.cgroupv2
+}
+
+// Destroy unmounts cgroupfs that were temporarily mounted
+func (cs *Cgroups) Destroy() error {
+	if cs.cgroupv1 != nil {
+		err := cs.cgroupv1.Destroy()
+		if err != nil {
+			return err
+		}
+	}
+	if cs.cgroupv2 != nil {
+		err := cs.cgroupv2.Destroy()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//
+// Cgroup
+//
+
+func NewCgroup(ver CgroupVersion) (*Cgroup, error) {
+	c := &Cgroup{}
+
+	var tmpPath, fsType, controller string
+
+	switch ver {
+	case CgroupV1:
+		tmpPath = TmpCgroupV1Path
+		fsType = CgroupV1FsType
+		controller = DefaultCgroupController
+	case CgroupV2:
+		tmpPath = TmpCgroupV2Path
+		fsType = CgroupV2FsType
+		controller = "" // no default controller for cgroupv2
+	default:
+		return c, failWrongCgroupVersion()
```

Comment:
considering we are returning a pointer, best to return nil if there is an error, no?
if you agree, it would make sense to change all returns of errors on this method

```suggestion
		return nil, failWrongCgroupVersion()
```